src/frontend/src/pages/ProjectDetailPage/ProjectForm/ProjectCreateChangeRequestForm.tsx

```diff
@@ -0,0 +1,270 @@
+/*
+ * This file is part of NER's FinishLine and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import {
+  Autocomplete,
+  AutocompleteRenderInputParams,
+  Box,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Grid,
+  MenuItem,
+  RadioGroup,
+  Select
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import NERSuccessButton from '../../../components/NERSuccessButton';
+import * as yup from 'yup';
+import { Controller, useFieldArray, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { ChangeRequestReason, ChangeRequestType } from 'shared';
+import { TextField, IconButton } from '@mui/material';
+import { wbsTester } from '../../../utils/form';
+import ReactHookTextField from '../../../components/ReactHookTextField';
+import { NERButton } from '../../../components/NERButton';
+import { ChangeEvent } from 'react';
+import { ProjectCreateChangeRequestFormInput } from './ProjectEditContainer';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { ProjectFormInput } from './ProjectForm';
+import { CreateChangeRequestFormInput } from '../../CreateChangeRequestPage/CreateChangeRequest';
+
+interface ProjectCreateChangeRequestFormProps {
+  onSubmit: (data: ProjectCreateChangeRequestFormInput) => void;
+  open: boolean;
+  onClose: () => void;
+  projectEdits?: ProjectFormInput;
+}
+
+const schema = yup.object().shape({
+  type: yup.string().required('Type is required'),
+  what: yup.string().required('What is required'),
+  why: yup
+    .array()
+    .min(1, 'At least one Why is required')
+    .required('Why is required')
+    .of(
+      yup.object().shape({
+        type: yup.string().required('Why Type is required'),
+        explain: yup
+          .string()
+          .required('Why Explain is required')
+          .when('type', {
+            is: ChangeRequestReason.OtherProject,
+            then: yup.string().test('wbs-num-valid', 'WBS Number is not valid', wbsTester)
+          })
+      })
+    )
+});
+
+const ProjectCreateChangeRequestForm: React.FC<ProjectCreateChangeRequestFormProps> = ({
```

Comment:
this shouldnt just be for projects, it should be for wbs elements, so we can use it with work packages too right @RChandler234